testsuite/tests/typing-local/effects.ml

```diff
@@ -0,0 +1,215 @@
+(* TEST
+   runtime5;
+   arch_amd64;
```

Comment:
Please leave a comment at the top of the arm64 emitter that it needs updating to support effects.  I'll probably do the updates themselves in due course.